tools/Makefile.unix

```diff
@@ -42,9 +42,13 @@ CONFIG_VERSION_BUILD ?= "0"
 VERSION_ARG = -v $(CONFIG_VERSION_STRING) -b $(CONFIG_VERSION_BUILD)
 else
 
-# Generate .version every time from GIT history
+# Generate .version.tmp every time from GIT history
+# Only update .version if the contents actually changes
+
+$(shell tools/version.sh .version.tmp)
+$(shell cp -u .version.tmp .version)
```

Comment:
why not use mv -u?

Reply:
I don't think the `-u` option is available for all platforms .

Reply:
Actually the macOS build is already complaining about it: https://github.com/apache/incubator-nuttx/runs/4664246522?check_suite_focus=true#step:6:812